codegen/main.cpp

```diff
@@ -0,0 +1,15 @@
+#include "javabind/codegen.hpp"
+#include <iostream>
+
+JNIEXPORT void codegen(const std::filesystem::path& output_path);
```

Comment:
I am not sure I understand the purpose of `JNIEXPORT`. Do we invoke this function directly from Java?

Reply:
JNIEXPORT will resolve to `__declspec( dllimport )` here on Windows and is needed to declare that the function is imported from a library. It might be confusing because we reuse JNIEXPORT. We could define JAVABIND_EXPORT as an alias for JNIEXPORT and use that for the codegen function?

Reply:
Even if it is somewhat more verbose to re-define this symbol, I would opt for doing so, and I would keep `JNIEXPORT ` exclusively for the interface between C++ and Java. (The code generator is an interface between a C++ library and a C++ executable.)

This change (and many others) have been pushed to the branch `master`. Specifically, I had to add `rebind_alloc` to make sure the allocator type is compatible with `boxed_t<T>` to eliminate a Visual Studio Community 2022 compiler error. I also had to reorganize the code a bit to make sure `print_registered_bindings` works as in the documentation. (It was triggering debug assertions when inserted back.) Other changes are mostly aesthetic (e.g. they reduce the number of symbols imported into public namespaces) and don't impact the spirit of your contribution.

For now, I have chosen to keep the manual native class/function declarations in Java for the sake of documentation purposes. However, the code generator has already helped add declarations that I have identified to be missing. It has already been a great utility!